src/umpire/resource/CudaConstantMemoryResource.cu

```diff
@@ -67,9 +67,13 @@ void CudaConstantMemoryResource::deallocate(void* ptr)
 {
   UMPIRE_LOG(Debug, "(ptr=" << ptr << ")");
 
-  util::AllocationRecord* record = ResourceManager::getInstance().deregisterAllocation(ptr);
+  auto record = ResourceManager::getInstance().deregisterAllocation(ptr);
```

Comment:
Question: can't deregisterAllocation return a copy of the object instead of a pointer?

Reply:
Yes, it could.

Reply:
I don't think it would be too much of a performance hit and that way, the callers would own the copied object.  And since it is an object instead of a object pointer, object deletion would be automatically deleted via scope